src/psyclone/tests/psyir/nodes/extract_node_test.py

```diff
@@ -47,7 +47,8 @@ def test_extract_node_equality():
     symboltable = SymbolTable()
     symboltable2 = SymbolTable()
     sched1 = Schedule(symbol_table=symboltable)
-    sched2 = Schedule(symbol_table=symboltable)
+    sched2 = Schedule()
+    sched2._symbol_table = symboltable  # Make sure it has the same ST instance
```

Comment:
I'm aware this is probably my Python ignorance speaking here but why do we need this?

Reply:
The equality operator of a ScopedNode (and subclasses of it) only returns True if the symbol tables are the exact same instance.

Since I changed the constructor to make a deepcopy of the provided symbol table if it already has an scope, this will never be true unless I manually inject the same instance.

Currently scopes will never be equal in user code. We could refine this by defining a equality operator for symbol tables, but if we need/want this I think should be discussed in a separate issue.

Reply:
In that case, I'd personally prefer the same construction for all schedules:
```
sched1 = Schedule()
sched1._symbol_table = symboltable
```

Reply:
Good point. I took you suggestion.